shm/inc/Delegate.hpp

```diff
@@ -0,0 +1,7 @@
+#pragma once
+
+class Delegate {
+public: 
+    virtual void payCrew(const size_t payCrew) = 0;
+    virtual ~Delegate() = 0; 
```

Comment:
Destructor doesn't have to be pure virtual but must be virtual. Class is already abstract as payCrew() is pure virtual.